sources/util.go

```diff
@@ -244,3 +246,32 @@ func mustParseInt(value string) int {
 }
 
 var accept = "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8"
+
+func unzipInMemory(ctx context.Context, body []byte) ([]byte, error) {
+	log := app.Log.From(ctx)
```

Comment:
Here's design flaw - you pass context just to get a logger from it. How logger is used? Log errors. That's wrong design for this lower level source specific utility.